cpp/src/aztec/ecc/curves/bn254/scalar_multiplication/scalar_multiplication.test.cpp

```diff
@@ -510,19 +510,10 @@ TEST(scalar_multiplication, endomorphism_split)
     fr* k2_t = (fr*)&scalar.data[2];
 
     fr::split_into_endomorphism_scalars(scalar, *k1_t, *k2_t);
-    // The compiler really doesn't like what we're doing here,
-    // and disabling the array-bounds error project-wide seems unsafe.
-    // The large macro blocks are here to warn that we should be careful when
-    // aliasing the arguments to split_into_endomorphism_scalars
-#if !defined(__clang__) && defined(__GNUC__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Warray-bounds"
-#endif
+
     fr k1{ (*k1_t).data[0], (*k1_t).data[1], 0, 0 };
     fr k2{ (*k2_t).data[0], (*k2_t).data[1], 0, 0 };
-#if !defined(__clang__) && defined(__GNUC__)
-#pragma GCC diagnostic pop
-#endif
+
```

Comment:
Can we just keep this in? I'm fine with not officially supporting GCC and removing it from the CI, but this error was occurring in the alpine1.17 gcc 12 build. Also, the thing we're doing here is very tricky and I think the textual warning should stay.